src/index.ts

```diff
@@ -35,3 +35,9 @@ export function validate(dl: string, states?: string | string[]) {
 
   return results.length ? results : null;
 }
+
+export function validateIgnoreCase(dl: string, states?: string | string[]) {
+  validate(dl.toUpperCase(), states);
```

Comment:
I think you need to see if you can invoke the regex test with the case insensitive switch instead of relying that all regex patterns match upper case.

Reply:
I Made changes for case insensitive regex match 
new RegExp(item.regex, 'i')